### ISSUE TITLE
[Feature Request] Build for Nintendo Switch

### ISSUE THREAD

**Issue by Nolij** [ipodtouch0218/NSMB-MarioVsLuigi#122]
It'd be hilarious if this were ported to the Switch, and assuming there's some way to build C# & Unity for the Switch it should be doable (not sure if that exists, but if it does it should be relatively straightforward).

**Comment by SLG64** [comment]
This would be a great idea, I would like it on NDSi tho.

**Comment by Nolij** [comment]
> This would be a great idea, I would like it on NDSi tho.

I think the difference there is that the Switch has a chance of running this at a playable framerate, whereas I heard my DSi scream in fear when I read your comment.

**Comment by SLG64** [comment]
Well, I wouldn't be surprised if the DSi software couldn't support it, seeing as how old/small it is. It would be a grand achievement if they got it to work on the DSi, and the Nintendo Switch would be a decent achievement.

**Comment by SLG64** [comment]
Exactly, which I don't understand, seeing as how this would be a major opportunity to get players back on a game they haven't played in a while. I have seen multiple fan-games get taken down, and some major gaming apps won't be on switch for that reason.

**Comment by Nolij** [comment]
> I would love to play this on switch, unfortunately Nintendo hates fan games and would 100% deny this.

I meant as Homebrew. Nintendo would 100% not allow this on the eShop, you are correct. They don't get any more say over Homebrew than they do PC though.

**Comment by SLG64** [comment]
I guess you're right. I don't understand why Nintendo wouldn't like to see what the community has to offer with creativity for new games.

**Comment by SLG64** [comment]
I guess that helps me understand

**Comment by ipodtouch0218** [comment]
Builds for Nintendo Switch require a switch SDK to create, which I don't have access to. Someone's made a switch version of the 1.5 update (since that uses an older Unity / SDK version), but nothing's planned for the current release.

**Comment by Nolij** [comment]
Where can I find that build?

**Comment by abso1utezer0** [comment]
This isn't a trademark issue, it is an sdk issue. This wouldn't be accepted to the E-Shop, *ever*. Even if someone did use a leaked sdk or misused an SDK granted to them by Nintendo to build for the Nintendo Switch (which would be quite illegal, and unless tracking was scrubbed Nintendo can see what you are using the SDK for), there is still the issue of getting the game to run on an actual Nintendo Switch. Last time I checked, there are only about ~13 million exploitable Nintendo Switches and the guy who was making and selling mod chips for the unhackable Switches was put in federal prison for a long time, so, yeah.

Basically: not worth the effort and time to target the Switch when like 5 people will use that version and a mobile port is a thing.

**Comment by Nolij** [comment]
> This isn't a trademark issue, it is an sdk issue. This wouldn't be accepted to the E-Shop, *ever*. Even if someone did use a leaked sdk or misused an SDK granted to them by Nintendo to build for the Nintendo Switch (which would be quite illegal, and unless tracking was scrubbed Nintendo can see what you are using the SDK for), there is still the issue of getting the game to run on an actual Nintendo Switch. Last time I checked, there are only about ~13 million exploitable Nintendo Switches and the guy who was making and selling mod chips for the unhackable Switches was put in federal prison for a long time, so, yeah.
> 
> Basically: not worth the effort and time to target the Switch when like 5 people will use that version and a mobile port is a thing.

I agree on the eShop, shouldn't even be tried. That being said, every Switch is exploitable via a modchip. TX chips are no longer available, but clones (which work quite well) are quite obtainable. Also, the mobile port doesn't let me play this game on my Switch.

**Comment by kiwiphoenix364** [comment]
I tried running this in the DNS hack which opens chrome (no soft/hard mods required, you can research how to do this), but when It bring up the website on it, it just shows a black window when hit run game. :( I don't think chrome for switch supports Unity, but it would be insane if it could (and have controller support, as PC does).

**Comment by abso1utezer0** [comment]
there is no "chrome for switch", the switch browser is some propriety browser contracted from a internet software company in japan and barely supports any html5 or webgl features so i'm not surprised

**Comment by kiwiphoenix364** [comment]
3ds homebrew port? (yes I know I can play the original on 3ds) I've seen unity games running on 3ds before.

**Comment by kiwiphoenix364** [comment]
> Well, I wouldn't be surprised if the DSi software couldn't support it, seeing as how old/small it is. It would be a grand achievement if they got it to work on the DSi, and the Nintendo Switch would be a decent achievement.

3ds/n3ds could be more feasible than DSI, and is still easy to homebrew. Idk how perf/online would be though.

**Comment by ripjaw1219** [comment]
you can use unity for the nintendo switch because the stanley parable ultra deluxe was made in unity and its on eshop

**Comment by EpicDoodleboy2** [comment]
> > Well, I wouldn't be surprised if the DSi software couldn't support it, seeing as how old/small it is. It would be a grand achievement if they got it to work on the DSi, and the Nintendo Switch would be a decent achievement.
> 
> 3DS/N3DS could be more feasible than DSI, and is still easy to homebrew. Idk how perf/online would be though.

The Normal 3ds (2ds/3dsXL) will not work because the original 3ds's can't run unity games but the 'New' 3ds CAN run unity games and nsmb VS could work with some optimisations. (It needs to work with 256 of ram tho)

**Comment by abso1utezer0** [comment]
can people please stop bumping this godawful request, it would be illegal to build to any of these platforms as they are all locked behind nda bound sdks

**Comment by Nolij** [comment]
> can people please stop bumping this godawful request, it would be illegal to build to any of these platforms as they are all locked behind nda bound sdks

The request is to do it as homebrew, which would not break any laws and not require violating any NDAs either.

**Comment by EpicDoodleboy2** [comment]
(New 3ds/2ds)
The nsmb VS Folder on PC has to be in the root on the SD CARD or the 3ds folder then they would install a .cia file as a forwarder for the homebrew.

Top Screen) NSMB VS logo
Icon) NSMB DS icon

**Comment by abso1utezer0** [comment]
> > can people please stop bumping this godawful request, it would be illegal to build to any of these platforms as they are all locked behind nda bound sdks
> 
> The request is to do it as homebrew, which would not break any laws and not require violating any NDAs either.

how exactly do you plan to build unity for switch or 3ds?

@ipodtouch0218 can you just close this good lord

**Comment by SLG64** [comment]
Can't the switch get games imported by an SD card? The Wii can import games from a memory card, so doesn't that make sense for the Switch? I don't see that it has any software for this to work though.

**Comment by Nolij** [comment]
> > Can't the switch get games imported by an SD card? The Wii can import games from a memory card, so doesn't that make sense for the Switch? I don't see that it has any software for this to work though.
> 
> I'd assume that's only for Switch homebrew, and as Nintendo still supports the Switch and there is a good chance of it getting bricked if homebrewed, unlike 3ds and Wii. That's why the modding community is so big on Wii/3ds compared to Switch. When Switch stops getting updates, maybe Switch homebrew could be possible. Switch SD cards are formatted to the Switch, so I'd assume that there is no way to download external software without a modded Switch.

Aside from "there is no way to download external software without a modded Switch" part there is very little accuracy in your reply. There is very little risk of bricking due to homebrew on the Switch. There are many people offering modchip installs who will replace your Switch if they damage it, and they are unlikely to damage it as it does not take obscene soldering skill to successfully mod most Switch models. I had very little soldering experience prior to modding mine, and haven't bricked one yet, after modding my OG, Lite, and OLED. Please don't spread misinformation about topics you clearly don't know very much about. If you want to argue this would be a small userbase, go ahead. But don't claim there's a high likelihood of getting bricked when that just isn't true.

**Comment by kiwiphoenix364** [comment]
> > > Can't the switch get games imported by an SD card? The Wii can import games from a memory card, so doesn't that make sense for the Switch? I don't see that it has any software for this to work though.
> > 
> > 
> > I'd assume that's only for Switch homebrew, and as Nintendo still supports the Switch and there is a good chance of it getting bricked if homebrewed, unlike 3ds and Wii. That's why the modding community is so big on Wii/3ds compared to Switch. When Switch stops getting updates, maybe Switch homebrew could be possible. Switch SD cards are formatted to the Switch, so I'd assume that there is no way to download external software without a modded Switch.
> 
> Aside from "there is no way to download external software without a modded Switch" part there is very little accuracy in your reply. There is very little risk of bricking due to homebrew on the Switch. There are many people offering modchip installs who will replace your Switch if they damage it, and they are unlikely to damage it as it does not take obscene soldering skill to successfully mod most Switch models. I had very little soldering experience prior to modding mine, and haven't bricked one yet, after modding my OG, Lite, and OLED. Please don't spread misinformation about topics you clearly don't know very much about. If you want to argue this would be a small userbase, go ahead. But don't claim there's a high likelihood of getting bricked when that just isn't true.

Sorry, I should have phrased it better. It takes more expertise (and hard modding usually) than the other consoles, where you can just download some software to an SD card (soft modding) and *if you decide to update your switch* there may be a slight chance of bricking. I'm not sure about that completely, since Idk how much brick protection modchips offer. I should have specified that I've never modded a switch (I have Wii and 3ds though), but that there are more risks to people who don't know technology well.

**Comment by SLG64** [comment]
Ok, that's more explanatory than last time. Thanks for clarifying.

**Comment by EpicDoodleboy2** [comment]
Also New 3ds is more easier to port nsmb vs because of the advantage of running unity files while the Wii didn't have official unity support but The Wii U is capible of running Unity games

Basicly saying that the Wii and O3DS might not be possible but the N3DS and Wii U Might be possible.

**Comment by tomrow** [comment]
Don't try building this for Nintendo consoles; Unity projects require official Nintendo development tools to build for Nintendo systems. Nintendo doesn't license its SDK for homebrew projects, and they'll c&d this project into next week if they find you've been using a stolen SDK.

**Comment by EpicDoodleboy2** [comment]
> Don't try building this for Nintendo consoles; Unity projects require official Nintendo development tools to build for Nintendo systems. Nintendo doesn't license its SDK for homebrew projects, and they'll c&d this project into next week if they find you've been using a stolen SDK.

I could of provided what I was saying,
I was saying that The Wii U and N3DS had Unity work for its platform and forgot to say that It could run the game via the NSMBVS PC Files on the sd card (like the RSDK decomps I used to port sonic 1 forever ver 1.4.2)
so it wouldn't require a nintendo dev unity account and a supported unity version because they removed 3ds on unity and bypass laws.
(and no, I don't have a dev account and unity)

I was Saying It could be possible if a enough time has passed, unity files might be able to run off the Sd card if someone has the passion to do it or run .exe files on those consoles which is impossible to my knowledge.

**Comment by WayloFunk** [comment]
Or maybe get switchroot android installed and play it from there.
